src/Symfony/Component/Security/Http/Event/LoginSuccessEvent.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Symfony\Component\Security\Http\Event;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * This event is dispatched after authentication has successfully completed.
+ *
+ * At this stage, the authenticator created an authenticated token
+ * and generated an authentication success response. Listeners to
+ * this event can do actions related to successful authentication
+ * (such as migrating the password).
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+class LoginSuccessEvent extends Event
+{
+    private $authenticator;
+    private $authenticatedToken;
+    private $request;
+    private $response;
+    private $providerKey;
+
+    public function __construct(AuthenticatorInterface $authenticator, TokenInterface $authenticatedToken, Request $request, ?Response $response, string $providerKey)
+    {
+        $this->authenticator = $authenticator;
+        $this->authenticatedToken = $authenticatedToken;
+        $this->request = $request;
+        $this->response = $response;
+        $this->providerKey = $providerKey;
+    }
+
+    public function getAuthenticator(): AuthenticatorInterface
+    {
+        return $this->authenticator;
+    }
+
+    public function getAuthenticatedToken(): TokenInterface
```

Comment:
WDYT about a `getUser(): ?UserInterface` convenience method?

Reply:
I'm in favor of not adding the method now. It would keep the door a bit open for replacing-user features in 5.2 - 5.4. If we don't get to it and `getUser()` is simply, we can always introduce it before removing experimental status

Reply:
lol, I just realized that I asked this question *again* today. I'm fine with not having it - especially if we stop dispatching this for anonymous login - implementing a listener is not that common.

Reply:
Wait, now I'm reconsidering. This `getUser(): UserInterface` wouldn't be on an interface - just am method in a concrete class. That's relatively easy to deprecate in the future compared to other things. I'm still +1 for including it for ease.